src/TitleBindPieces/TitleBindPiecesContainer/TitleBindPiecesContainer.js

```diff
@@ -0,0 +1,149 @@
+  const bindPieceData = useRef(null);
+
+  const { bindPieces, isBinding } = useBindPiecesMutation();
+
+  const {
+    isLoading,
+    holdingLocations,
+    orderLine,
+    pieces = [],
+    title,
+  } = useTitleHydratedPieces({
+    titleId,
+    receivingStatus: PIECE_STATUS.received,
+    searchQuery: 'isBound==false',
+  });
+
+  const onCancel = useCallback(() => {
+    history.push({
+      pathname: `/receiving/${titleId}/view`,
+      search: location.search,
+    });
+  }, [history, titleId, location.search]);
+
+  const bindItems = (requestData) => {
+    return bindPieces(requestData)
+      .then(() => {
+        onCancel();
+        showCallout({ messageId: 'ui-receiving.bind.pieces.create.success' });
+      }).catch(() => {
+        showCallout({
+          type: 'error',
+          messageId: 'ui-receiving.bind.pieces.create.error',
+        });
+      });
+  };
+
+  const onConfirm = (requestsAction) => {
+    toggleOpen();
+
+    if (requestsAction === TRANSFER_REQUEST_ACTIONS.cancel) {
+      setShowDeleteMessage(false);
+
+      return null;
+    }
+
+    return bindItems({
+      ...bindPieceData.current,
+      requestsAction,
+    });
+  };
+
+  const onSubmit = async (values) => {
+    const selectedItems = values.receivedItems.filter(({ checked }) => checked);
+    const openRequests = selectedItems.filter(({ itemId, request }) => itemId && request);
+
+    const requestData = {
+      poLineId: orderLine.id,
+      bindPieceIds: selectedItems.map(({ id }) => id),
+      bindItem: {
+        ...values.bindItem,
+      },
+    };
+
+    if (openRequests?.length) {
+      const hasDifferentRequesterId = openRequests.some(({ request }) => request?.requesterId !== currentTenantId);
+
+      setShowDeleteMessage(hasDifferentRequesterId);
+      bindPieceData.current = requestData;
+      toggleOpen();
+    } else {
+      bindItems(requestData);
+    }
+  };
+
+  if (isLoading) {
+    return (
+      <Paneset>
+        <LoadingPane />
+      </Paneset>
+    );
+  }
+
+  const initialValues = { receivedItems: pieces };
+  const paneTitle = `${orderLine?.poLineNumber} - ${title?.title}`;
+
+  return (
+    <>
+      <TitleBindPieces
+        initialValues={initialValues}
+        onCancel={onCancel}
+        onSubmit={onSubmit}
+        paneTitle={paneTitle}
+        instanceId={title.instanceId}
+        locations={holdingLocations}
+        isLoading={isBinding}
+      />
+      <TitleBindPiecesConfirmationModal
+        id="confirm-binding-modal"
+        onCancel={toggleOpen}
+        onConfirm={onConfirm}
+        showDeleteMessage={showDeleteMessage}
+        open={open}
+      />
+    </>
+  );
+}
+
+TitleBindPiecesContainer.propTypes = {
+  history: ReactRouterPropTypes.history.isRequired,
+  location: ReactRouterPropTypes.location.isRequired,
+  match: ReactRouterPropTypes.match.isRequired,
+  stripes: stripesShape.isRequired,
+};
+
+export default stripesConnect(TitleBindPiecesContainer);
```

Comment:
you don't need `stripesConnect`. If you need the `stripes` object use `useStripes` hook